app/controllers/api/v3/providers_controller.rb

```diff
@@ -7,7 +7,8 @@ def index
         build_fields_for_index
 
         @providers = if params[:search].present?
-                       @recruitment_cycle.providers.search_by_code_or_name(params[:search])
+                       normalise_query = CGI.unescape(params[:search]).downcase.gsub(/[^0-9a-z]/i, "")
```

Comment:
Similar as above about whether we need `CGI.unescape` at all, and separation of responsibilities.`